pkg/cri/sbserver/podsandbox/sandbox_run.go

```diff
@@ -210,12 +223,12 @@ func (c *Controller) Start(ctx context.Context, id string) (_ *api.ControllerSta
 		return nil, fmt.Errorf("failed to create containerd task: %w", err)
 	}
 	defer func() {
-		if retErr != nil {
+		if retErr != nil && cleanupErr == nil {
 			deferCtx, deferCancel := ctrdutil.DeferContext()
 			defer deferCancel()
 			// Cleanup the sandbox container if an error is returned.
-			if _, err := task.Delete(deferCtx, WithNRISandboxDelete(id), containerd.WithProcessKill); err != nil && !errdefs.IsNotFound(err) {
-				log.G(ctx).WithError(err).Errorf("Failed to delete sandbox container %q", id)
+			if _, cleanupErr = task.Delete(deferCtx, WithNRISandboxDelete(id), containerd.WithProcessKill); cleanupErr != nil && !errdefs.IsNotFound(cleanupErr) {
```

Comment:
Similar to https://github.com/containerd/containerd/pull/5904#discussion_r979136252, cleanupErr shouldn't be set if `task.Delete` returns NotFound error.

I propose:

```
			if _, err := task.Delete(deferCtx, WithNRISandboxDelete(id), containerd.WithProcessKill); err != nil && !errdefs.IsNotFound(err) {
				log.G(ctx).WithError(cleanupErr).Errorf("Failed to delete sandbox container %q", id)
				cleanupErr = err
			}
```

Reply:
Done.